### ISSUE TITLE
New service: AWS Lake Formation

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#9700]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Lake Formation is now GA.

### New or Affected Resource(s)

* aws_XXXXX

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/08/aws-lake-formation-is-now-generally-available/).
[Blog post](https://aws.amazon.com/blogs/aws/aws-lake-formation-now-generally-available/).

Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/9670

**Comment by slapula** [comment]
I'm also looking into writing up these resources as well.

**Comment by rafiparvez** [comment]
We are using AWS lakeformation for creating data lake at my organization. I am interested in contributing to this.

**Comment by ewbankkit** [comment]
@rafiparvez Could you please propose some resources that would match your uses cases?

**Comment by dthauvin** [comment]
@ewbankkit one off major features that brings AWS LakeFormation , is the capabilities to control fine-grained privilege on GlueTable ( like GRANT at column level) and s3 location with course-grained Permissions. So far to enable use case around fine grained i think we need 3 ressources:  

AWS::LakeFormation::DataLakeSettings ( config between IAM policy and LakeFormation policy) 
AWS::LakeFormation::Permissions (represents the permissions that a principal has on an AWS)
AWS::LakeFormation::Resource (represents the data (Amazon S3 buckets and folders)

BR

**Comment by gmazelier** [comment]
Here is a proposal to support Lake Formation settings (admins configuration), resources (de/register data locations) and permissions. Feedbacks are welcome!

**Comment by spaszek** [comment]
Any ETA?

**Comment by packetlag** [comment]
bump for an update, please

**Comment by ewbankkit** [comment]
@packetlag This has been added to the [August - October 2020 Roadmap](https://github.com/terraform-providers/terraform-provider-aws/blob/master/ROADMAP.md#aws-lake-formation). Thanks.

**Comment by peramnaveen29** [comment]
#14411
I guess it is related with the current AWS Lake Formation enhancement

**Comment by theophilechevalier** [comment]
Hi @ewbankkit,
I'm willing to finish the implementation but there are already numerous open pull requests on this matter. Can you plan a review so we can fix what's missing? We're a lot to need this feature

**Comment by roberth-k** [comment]
Resource Links (https://github.com/terraform-providers/terraform-provider-aws/issues/15296) and CreateTableDefaultPermissions (https://github.com/terraform-providers/terraform-provider-aws/issues/14411) should be implemented as part of this epic, as they are essential for accessing cross-account Lake Formation resources using Athena and Redshift Spectrum.

**Comment by YakDriver** [comment]
The first release with Lake Formation support is likely to these data sources and resources:

- [ ] resource/aws_lakeformation_resource
- [ ] data source/aws_lakeformation_resource
- [ ] resource/aws_lakeformation_settings
- [ ] resource/aws_lakeformation_permissions
- [ ] data source/aws_lakeformation_permissions

Subsequent releases (including additional contributions from community members) may include these data sources and resources:

* data source/aws_lakeformation_resources
* data source/aws_lakeformation_settings
* data source/aws_lakeformation_effective_permissions

**Comment by priyen** [comment]
@YakDriver Do you have any idea of any sort of ETA for lake formation support being available? I see lots of in flight PRs above.

**Comment by YakDriver** [comment]
Should be available with version 3.22!

**Comment by YakDriver** [comment]
Closing this since significant Lake Formation funcitonality has been merged. Remaining functionality is identified in #16845 and #16846. If there are other issues and features needed, please open a new issue! Thank you for your interest in Lake Formation!
